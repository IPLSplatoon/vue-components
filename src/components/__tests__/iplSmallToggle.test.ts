import IplSmallToggle from '../iplSmallToggle.vue';
import { mount } from '@vue/test-utils';

describe('IplSmallToggle', () => {
    it('matches snapshot when toggle value is true', () => {
        const wrapper = mount(IplSmallToggle, {
            props: {
                label: 'Toggle!',
                modelValue: true
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot when toggle value is false', () => {
        const wrapper = mount(IplSmallToggle, {
            props: {
                label: 'Toggle!',
                modelValue: false
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot when toggle is disabled', () => {
        const wrapper = mount(IplSmallToggle, {
            props: {
                label: 'Toggle!',
                modelValue: true,
                disabled: true
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('emits event in inner input change', () => {
        const wrapper = mount(IplSmallToggle, {
            props: {
                label: 'Toggle!',
                modelValue: true,
                disabled: false
            }
        });

        const input = wrapper.find('input');
        input.element.checked = false;
        input.element.dispatchEvent(new Event('change'));

        const emitted = wrapper.emitted('update:modelValue');
        expect(emitted?.length).toEqual(1);
        expect(emitted?.[0]).toEqual([false]);
    });

    it('does nothing when clicking wrapper while toggle is disabled', async () => {
        const wrapper = mount(IplSmallToggle, {
            props: {
                label: 'Toggle!',
                modelValue: true,
                disabled: true
            }
        });

        await wrapper.get('.ipl-small-toggle__wrapper').trigger('click');

        expect(wrapper.emitted('update:modelValue')).toBeFalsy();
    });
});
