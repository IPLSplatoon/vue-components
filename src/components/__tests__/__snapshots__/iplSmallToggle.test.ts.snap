// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IplSmallToggle matches snapshot when toggle is disabled 1`] = `
"<div class=\\"ipl-small-toggle__wrapper disabled active\\">Toggle! <div class=\\"value-display\\">
    <div class=\\"circle\\"></div>
  </div>
</div>"
`;

exports[`IplSmallToggle matches snapshot when toggle value is false 1`] = `
"<div class=\\"ipl-small-toggle__wrapper\\">Toggle! <div class=\\"value-display\\">
    <div class=\\"circle\\"></div>
  </div>
</div>"
`;

exports[`IplSmallToggle matches snapshot when toggle value is true 1`] = `
"<div class=\\"ipl-small-toggle__wrapper active\\">Toggle! <div class=\\"value-display\\">
    <div class=\\"circle\\"></div>
  </div>
</div>"
`;
